src/App.tsx

```diff
@@ -1,26 +1,218 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { FormEvent, useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import {
+  USER_ID,
+  creatTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+} from './api/todos';
+import { Status } from './types/Status';
+import { Todo } from './types/Todo';
+import { Error } from './types/Error';
 
-const USER_ID = 0;
+import { Header } from './conponents/Header/Header';
+import { TodoList } from './conponents/TodoList/TodoList';
+import { Footer } from './conponents/Footer/Footer';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState<Status>(Status.All);
+  const [error, setError] = useState<string>('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodos, setLoadingTodos] = useState<number[]>([]);
+  const [newTitle, setNewTitle] = useState<string>('');
+
+  const titleField = useRef<HTMLInputElement>(null);
+
+  const completedTodos = todos.filter(todo => todo.completed);
+  const activeTodos = todos.filter(todo => !todo.completed);
```

Comment:
This is ineffective, as on each render you call the filter two times, regardless of whether you need it or not.

I suggest you write some function (say `filterTodos`) that will perform filtering only based on the currently selected filters - or not perform the filtering at all.

```suggestion
  const filteredTodos = filterTodos(todos, filterBy);
```